### PR TITLE
fix: Check identity providers for oauth instead of check amount users.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
Check identity providers for oauth instead of check amount users.

### Referenced issue:
https://github.com/eclipse/che/issues/16836

### CODE REVIEW

File: pkg/controller/che/che_controller.go [hunk 1]
```diff
@@ -377,29 +381,33 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
...
-					helpLink = howToCreateAUserLinkOS3
+					helpLink = howToAddIdentityProviderLinkOS3
 				}
-				logrus.Errorf(failedNoOpenshiftUserMessage)
-				if err := r.SetStatusDetails(instance, request, failedNoOpenshiftUserReason, failedNoOpenshiftUserMessage, helpLink); err != nil {
-					return reconcile.Result{}, err
+				logrus.Warn(warningNoIdentityProvidersMessage, " ", helpLink)
+				instance.Spec.Auth.OpenShiftoAuth = false
+				if err := r.UpdateCheCRSpec(instance, "OpenShiftoAuth", strconv.FormatBool(instance.Spec.Auth.OpenShiftoAuth)); err != nil {
+					if errors.IsConflict(err) {
```
Comment (tolusha) [inline pkg/controller/che/che_controller.go:403]: In which cases the conflict is possible?

File: pkg/controller/che/che_controller.go [hunk 2]
```diff
@@ -377,29 +381,33 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
...
+						logrus.Warn("Conflict while getting Openshift oauth info. Updating Che cluster info to resolve conflict...")
```
Comment (tolusha) [inline pkg/controller/che/che_controller.go:404]: `getting` is mentioned but we are updating the CR, is it correct?

File: pkg/controller/che/che_controller.go [hunk 3]
```diff
@@ -377,29 +381,33 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
...
+						instance, _ = r.GetCR(request)
+						return reconcile.Result{}, err
+					} else {
+						logrus.Errorf("Failed to get OpenShift oauth info")
```
Comment (tolusha) [inline pkg/controller/che/che_controller.go:408]: getting is mentioned but we are updating the CR, is it correct?

File: pkg/controller/che/che_controller.go [hunk 4]
```diff
@@ -377,29 +383,35 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
...
+					getUsersErrMsg := failedUnableToGetOpenshiftUsers + " Cause: " + err.Error()
+					logrus.Errorf(getUsersErrMsg)
+					if err := r.SetStatusDetails(instance, request, failedNoOpenshiftUserReason, getUsersErrMsg, ""); err != nil {
+						return reconcile.Result{}, err
+					}
 					return reconcile.Result{}, err
 				}
-				return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+				if len(users.Items) < 1 {
+					logrus.Warn(warningNoIdentityProvidersMessage, " ", howToConfigureOAuthLinkOS3)
```
Comment (tolusha) [inline pkg/controller/che/che_controller.go:412]: It should be no users messages with corresponding link how to add user

**Comment by tolusha** [comment]
To fix olm tests it is needed to rebase on master

**Comment by AndrienkoAleksandr** [comment]
Tested with crc, openshift, minishift.

**Comment by tolusha** [comment]
Tested on OpenShift with and without oAuth Providers

**Comment by AndrienkoAleksandr** [comment]
Tested with minishift. Works well and stable. Also tested moving from enabled oAuth to disabled and otherwise.

**Pull request merged**